### PR TITLE
PM-10393 SSH keys

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10393

QA ticket:
https://bitwarden.atlassian.net/browse/PM-12729

(Server PR):
https://github.com/bitwarden/server/pull/4575

## 📔 Objective

This PR merges the feature branch for SSH keys to main.
All individual commits (aside from rebases) have been previously reviewed and approved in their PR's; but they have not been QA tested. **This PR will be QA tested**

Creation of SSH keys is gated behind a feature flag; use of them is not.

Included are:

SSH key item type:
- https://github.com/bitwarden/clients/pull/10360
- https://github.com/bitwarden/clients/pull/11046

Import export:
- https://github.com/bitwarden/clients/pull/10529
- https://github.com/bitwarden/clients/pull/11048

SSH Agent:
- https://github.com/bitwarden/clients/pull/10293
- https://github.com/bitwarden/clients/pull/10293
- https://github.com/bitwarden/clients/pull/11148

> [!NOTE]  
> Ssh key generation UI support will come in a follow-up PR tracked here: https://github.com/bitwarden/clients/pull/11232

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by djsmith85 — DISMISSED** [review]
Tools changes are looking good

**Review by coroiu — DISMISSED** [review]
Approving without review since I'm assuming the changes have already been reviewed in the sub-PRs

**Review by gbubemismith — DISMISSED** [review]

**Comment by quexten** [comment]
Resolved merge conflicts.

**Review by djsmith85 — APPROVED** [review]

**Review by coroiu — APPROVED** [review]
Approving without review

**Review by gbubemismith — DISMISSED** [review]

**Comment by quexten** [comment]
Moving to draft until QA passed and merge conflicts are resolved, as to keep re-reviews low.

**Comment by quexten** [comment]
Since the last approval, a bug found by QA was fixed and a feature flag added (the agent is now entirely feature flagged), and the PR's for this were reviewed and approved seperately.

**Review by gbubemismith — APPROVED** [review]

**Review by coroiu — APPROVED** [review]

File: apps/browser/src/vault/popup/components/vault/vault-items.component.ts [hunk 1]
```diff
@@ -106,6 +106,9 @@ export class VaultItemsComponent extends BaseVaultItemsComponent implements OnIn
           case CipherType.SecureNote:
             this.groupingTitle = this.i18nService.t("secureNotes");
             break;
+          case CipherType.SshKey:
+            this.groupingTitle = this.i18nService.t("sshKeys");
```
Comment (willmartian) [inline apps/browser/src/vault/popup/components/vault/vault-items.component.ts:110]: @quexten Isn't doesn't look like this i18n key exists in browser's `messages.json`

Comment (quexten) [reply apps/browser/src/vault/popup/components/vault/vault-items.component.ts:110]: https://github.com/bitwarden/clients/pull/12083

Thanks for catching it

Comment (willmartian) [reply apps/browser/src/vault/popup/components/vault/vault-items.component.ts:110]: Of course, and ty for fixing it!

**Pull request merged**